torch/csrc/autograd/engine.cpp

```diff
@@ -726,10 +715,31 @@ auto Engine::execute(const edge_list& roots,
   // Lock post_callbacks_lock_ before clearing final_callbacks_
   ClearCallbacks _cb_guard(final_callbacks_, post_callbacks_lock_);
 
+  bool is_reentrant_call = worker_device != NO_DEVICE;
+  std::shared_ptr<ReadyQueue> memorized_cpu_ready_queue = nullptr;
+
+  if (is_reentrant_call) {
```

Comment:
The logic looks a bit weird here.
maybe just do if this is not reentrant, initialize the ready queue.
Then when creating the graph_task just give `is_reentrant_call` and `local_ready_queue`.

Or am I missing something?